File: frontend/src/types/seat.ts
export type SeatState = 'available' | 'selected' | 'locked' | 'sold'

export interface SeatMapConfig {
  rows: string[]
  seats_per_row: number
}

export interface SeatItem {
  label: string
  row: string
  number: number
  state: SeatState
}

export interface SeatAvailabilityResponse {
  showtime_id: number
  movie_title: string
  cinema_name: string
  cinema_location: string
  room_name: string
  showtime_start: string
  showtime_end: string
  format: string
  language: string
  price_per_seat: string
  seat_map: SeatMapConfig
  seats: SeatItem[]
  selected_seat_labels: string[]
  countdown_seconds: number
  lock_expires_at: string | null
}

export interface SeatLockPayload {
  showtime_id: number
  seat_labels: string[]
  session_id: string
}

File: backend/app/schemas/seat.py
from datetime import datetime
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel, Field


SeatState = Literal["available", "selected", "locked", "sold"]


class SeatMapConfig(BaseModel):
    rows: list[str]
    seats_per_row: int = Field(gt=0)


class SeatItem(BaseModel):
    label: str
    row: str
    number: int
    state: SeatState


class SeatAvailabilityResponse(BaseModel):
    showtime_id: int
    movie_title: str
    cinema_name: str
    cinema_location: str
    room_name: str
    showtime_start: datetime
    showtime_end: datetime
    format: str
    language: str
    price_per_seat: Decimal
    seat_map: SeatMapConfig
    seats: list[SeatItem]
    selected_seat_labels: list[str]
    countdown_seconds: int
    lock_expires_at: datetime | None


class SeatLockRequest(BaseModel):
    showtime_id: int
    seat_labels: list[str] = Field(min_length=1)
    session_id: str = Field(min_length=8, max_length=120)


class SeatReleaseRequest(BaseModel):
    showtime_id: int
    seat_labels: list[str] = Field(min_length=1)
    session_id: str = Field(min_length=8, max_length=120)
